prelude/prelude.mlw

```diff
@@ -209,11 +209,9 @@ module Char
 end
 
 module Ghost
-  type ghost_ty 't
-  val function new (ghost x : 't) : ghost_ty 't
-  val function inner (x : ghost_ty 't) : 't
-  axiom new_spec: forall x: 't [new x]. inner (new x) = x
-  axiom inner_spec: forall x: ghost_ty 't [inner x]. new (inner x) = x
+  type ghost_ty 't = { ghost inner2 : 't }
+  let function new [@inline:trivial] (ghost x : 't) = { inner2 = x; }
+  let ghost function inner [@inline:trivial] (x : ghost_ty 't) = x.inner2
```

Comment:
`ghost_ty` was intentionally made an opaque type in #823 to fix the issue with infinite values in #436.

Reply:
Sorry, maybe I'm missing something, but I thought the goal of #823 was to prevent unsound recursive type definitions involving `Ghost` in Rust, and I don't see how that is related to the way `Ghost` is encoded in Why3.

Reply:
The check on the Rust level does not catch all unsound type definitions. For example, the following works:
```rust
struct Foo<T>(Ghost<T>);
struct Bad(Foo<Bad>);
```
To still prevent such unsound types, we additionally rely on the strict positivity check of Why3.
As `Ghost` is defined as opaque type, the parameter `T` of `Foo` is not considered strictly positive and thus the definition of `Bad` is rejected.